timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/api/Client.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.auth.api;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface Client {
+    Client ANONYMOUS = Client.regular("anonymous");
+
+    String getUserName();
+    boolean isAdmin();
```

Comment:
I think we should probably standardise the naming here and in the factories below, the `userName` field can probably just be `name` (or `clientName`, if you prefer)

Also, nit: we typically name the components in immutable classes as just the components (i.e. `userName` and `admin`); these getters should be autogenerated if I remember correctly.